### ISSUE TITLE
Broken shiny  app in burden-model.R

### ISSUE THREAD

**Issue by spleonard1** [barricklab/igem2019#3]
Hey Ginny! I found several bugs in the shiny code. Here's what you should do to fix them:

1) The shiny input function relies on several variables that are not defined in the script, specifically "u" and "b". These need to be defined in the script, or the input function changed to use constant values instead of variables.

2) Similarly, the volume part of input uses variables "vol" "vol2k"...etc but these are only defined within the main plot.ode.mod function, so they are not actually available to use in the input function. Move the definition of these variables outside the plot.ode.mod function.

3) The "differing number of rows" error is because the server function tries to build a data frame for the parameters (pars), but by default there is no volume selected/defined. So it throws an error when you build a data frame with a missing value. You can change this by having a default volume selected, or changing how the parameter inputs are handled.

4) Building off 3), I *really* think you should actually split the plot.ode.mod function into different parts. Instead of having one big function that does everything (and you have to pay attention to how input parameters are structured), I would split the function into smaller parts that a) calculate the necessary ODE values and b) plot the results. This simplifies the input you have to pass from the ui function, and maybe you can reuse the plotting functionality with different inputs. Even after you fix 1-3 above you'll get errors from how you're currently passing in the parameters to the ODE calculation. Since you don't need to pass in "volumes" in the parameter data.frame (they are just lines on the chart), you can remove that part.

Do these make sense? Take a crack at them at let me know if you need more advice / help. Good luck!

**Comment by ginnymortensen** [comment]
These made sense and I fixed the issues with the parameters, volumes, etc. I also changed the way the pars is passed into the solver and I split the function into different bits (excluded the plotting function).
However, now when I run the code, I get the error "object y not found". I think this is related to the ode function solver throwing an error because y is duplicated within it (the as.list function arranging the equations in the solver). I don't know how to fix this without compromising the solver's functionality outside of the shiny. Help?
